myst_nb/nb_glue/domain.py

```diff
@@ -0,0 +1,277 @@
+import copy
+import json
+from pathlib import Path
+from typing import List, Dict
+
+from docutils import nodes
+from docutils.parsers.rst import directives
+from sphinx.domains import Domain
+from sphinx.util.docutils import SphinxDirective
+from sphinx.util import logging
+
+
+from myst_nb.nb_glue import GLUE_PREFIX
+from myst_nb.nb_glue.utils import find_all_keys
+
+SPHINX_LOGGER = logging.getLogger(__name__)
+
+
+class PasteNode(nodes.container):
+    """Represent a MimeBundle in the Sphinx AST, to be transformed later."""
+
+    def __init__(self, key, location=None, rawsource="", *children, **attributes):
+        self.location = location
+        attributes["key"] = key
+        super().__init__("", **attributes)
+
+    @property
+    def key(self):
+        return self.attributes["key"]
+
+    def copy(self):
+        return self.__class__(location=self.location, **self.attributes)
+
+    def create_node(self, output: dict):
+        """Create the output node, give the cell output."""
+        # the whole output chunk is deposited and rendered later
+        # TODO move these nodes to separate module, to avoid cyclic imports
```

Comment:
I hope that we can just import these nodes from jupyter-sphinx at some point, whenever I can figure out the tests in https://github.com/jupyter/jupyter-sphinx/pull/107